web/src/stream_popover.js

```diff
@@ -597,6 +612,48 @@ export async function build_move_topic_to_stream_popover(
         $("#move_topic_modal .move_messages_edit_topic").on("input", () => {
             update_submit_button_disabled_state(stream_widget_value);
         });
+
+        let default_message_move_count;
+        // The "if" part covers the case when "Move topic" is clicked,
+        // while the "else" part accounts for the case when "Move messages" is clicked.
+        if (args.from_message_actions_popover) {
+            default_message_move_count =
+                message_util.get_count_of_messages_in_topic_sent_after_current_message(
+                    current_stream_id,
+                    topic_name,
+                    message.id,
+                );
+        } else {
+            default_message_move_count = message_util.get_messages_in_topic(
+                current_stream_id,
+                topic_name,
+            ).length;
+        }
```

Comment:
It seems fragile that this conditional needs to exactly match some other conditional for what the initial value is; can you align this to the other code?

Reply:
Ideally you'd just call `get_count_of_messages_to_be_moved` here as well...